### PR TITLE
Add Singular's implementation of King's algorithm

### PR DESCRIPTION
**Pull request by joschmitt**
Here is an interface to `invariant_ring_reynolds` in Singular as requested by @wdecker implementing King's algorithm to compute fundamental invariants.
I have the impression that this is considerably faster than the reduction of the primary and secondary invariants if the group is not too small (as expected).
Note that there is also `invariant_ring_perm` in Singular for permutation groups, but so far we do not support permutation groups at all. Should I also add this function?

### CODE REVIEW

**Comment by thofma** [comment]
I would keep the permutations group out for the moment.

File: experimental/InvariantTheory/invariant_rings.jl [hunk 1]
```diff
@@ -716,21 +716,42 @@ AbstractAlgebra.Generic.MatSpaceElem{nf_elem}[[0 0 1; 1 0 0; 0 1 0], [1 0 0; 0 a
...
 
+function fundamental_invariants_via_minimal_subalgebra(IR::InvRing)
+  V = primary_invariants(IR)
+  append!(V, irreducible_secondary_invariants(IR))
+  return minimal_subalgebra_generators(V)
+end
+
+function fundamental_invariants_via_king(IR::InvRing)
+  rey = reynolds_via_singular(IR)
+  F = Singular.LibFinvar.invariant_algebra_reynolds(rey)
```
Comment (tthsqe12) [inline experimental/InvariantTheory/invariant_rings.jl]: If you know the return typeof `LibFinvar.invariant_algebra_reynolds`, the compiler would probably thank you for a type assertion here, as it has absolutely no idea what is going to be returned.

Comment (joschmitt) [reply experimental/InvariantTheory/invariant_rings.jl]: Good point.
I fixed it a little bit, but getting the correct type is difficult...

Comment (tthsqe12) [reply experimental/InvariantTheory/invariant_rings.jl]: This is fine now.

**Pull request merged**